src/simtools/simtel/simtel_table_reader.py

```diff
@@ -298,6 +298,23 @@ def _adjust_columns_length(rows, n_columns):
     return [row[:n_columns] + [0.0] * max(0, n_columns - len(row)) for row in rows]
 
 
+def _process_line_parts(parts):
+    """Process a single line of data and convert parts to floats."""
+    row = []
+    i = 0
+    while i < len(parts):
+        try:
+            # Try to convert to float, if fails, it might be a string
+            row.append(float(parts[i]))
+        except ValueError:
+            logger.debug(f"Skipping non-float part: {parts[i]}")
+            # Skip this part and move to the next
```

Comment:
Are these lines necessary? Seems we anyway do only i+1 when leaving the except block.